### PR TITLE
provider/gce: Implemented bucket & object ACLs, as well as documentation and tests

### PR DESCRIPTION
**Pull request by lwander**
I pulled `predefined_acl` out of GCE buckets and objects to avoid issues with `force_new` described [here](https://github.com/hashicorp/terraform/issues/2289).

### CODE REVIEW

**Comment by lwander** [comment]
@sparkprime

File: builtin/providers/google/resource_storage_bucket_acl_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	"math/rand"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+
+	//"google.golang.org/api/storage/v1"
+)
+
+var roleEntityBasic1 = &RoleEntity{
```
Comment (sparkprime) [inline builtin/providers/google/resource_storage_bucket_acl_test.go]: single string is more concise

File: builtin/providers/google/resource_storage_bucket_acl_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				Check:  resource.ComposeTestCheckFunc(
+					testAccCheckGoogleStorageBucketAcl(testAclBucketName, roleEntityBasic1),
+					testAccCheckGoogleStorageBucketAcl(testAclBucketName, roleEntityBasic2),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGoogleStorageBucketAcl_update(t *testing.T) {
```
Comment (sparkprime) [inline builtin/providers/google/resource_storage_bucket_acl_test.go]: also test forcenew of the acl, and removing acl leaving bucket alive

File: builtin/providers/google/resource_storage_bucket_acl.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+	split := strings.Split(role_entity, ":")
+	if len(split) != 2 {
+		return nil, fmt.Errorf("Error, each role entity pair must be " +
+			"formatted as ROLE:entity")
+	}
+
+	return &RoleEntity{Role: split[0], Entity: split[1]}, nil
+}
+
+func resourceStorageBucketAclCreate(d *schema.ResourceData, meta interface{}) error {
```
Comment (sparkprime) [inline builtin/providers/google/resource_storage_bucket_acl.go]: To allow interop with manually managed acls, make semantics like the metadata:  Only read and delete entities that are mentioned in the terraform config.

File: builtin/providers/google/resource_storage_object_acl_test.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+						testAclObjectName, roleEntityBasic1),
+					testAccCheckGoogleStorageObjectAcl(testAclBucketName,
+						testAclObjectName, roleEntityBasic2),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGoogleStorageObjectAcl_update(t *testing.T) {
```
Comment (sparkprime) [inline builtin/providers/google/resource_storage_object_acl_test.go]: upgrade

File: builtin/providers/google/resource_storage_object_acl_test.go [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+		if (res.Role != roleEntity.Role) {
+			return fmt.Errorf("Error, Role mismatch %s != %s", res.Role, roleEntity.Role)
+		}
+
+		return nil
+	}
+}
+
+func testAccGoogleStorageObjectAclDestroy(s *terraform.State) error {
```
Comment (sparkprime) [inline builtin/providers/google/resource_storage_object_acl_test.go]: all you're testing is that the object is gone (and bucket for the other file), why not use the test from the bucket / object resource that checks that it is gone

**Comment by lwander** [comment]
@sparkprime

**Comment by sparkprime** [comment]
squash and then i'll merge

**Pull request merged**